web/impact/impact/v1/views/base_list_view.py

```diff
@@ -42,31 +42,27 @@ def get(self, request):
 
     def _results(self, limit, offset):
         queryset = self.helper_class.all_objects()
-        updated_at_gt = self.request.query_params.get('updated_at__gt', None)
-        updated_at_lt = self.request.query_params.get('updated_at__lt', None)
-        if updated_at_gt or updated_at_lt:
-            queryset = _filter_by_date(queryset, updated_at_gt, updated_at_lt)
+        updated_at_after = self.request.query_params.get(
+            'updated_at.after', None)
+        updated_at_before = self.request.query_params.get(
+            'updated_at.before', None)
+        if updated_at_after or updated_at_before:
+            queryset = _filter_by_date(queryset,
+                                       updated_at_after,
+                                       updated_at_before)
         count = queryset.count()
         return (count,
                 [self.helper_class(obj).serialize()
                  for obj in queryset[offset:offset + limit]])
 
 
-def _filter_by_date(queryset, updated_at_gt, updated_at_lt):
-    updated_at_gt = parse_date(updated_at_gt)
-    updated_at_lt = parse_date(updated_at_lt)
-    if updated_at_lt:
-        queryset = queryset.filter(
-            Q(updated_at__isnull=False)
-        ).exclude(
-            Q(updated_at__gte=updated_at_lt)
-        )
-    if updated_at_gt:
-        queryset.filter(
-            Q(updated_at__isnull=False)
-        ).exclude(
-            Q(updated_at__lte=updated_at_gt)
-        )
+def _filter_by_date(queryset, updated_at_after, updated_at_before):
+    updated_at_after = parse_date(updated_at_after)
+    updated_at_before = parse_date(updated_at_before)
+    if updated_at_after:
+        queryset = queryset.filter(Q(updated_at__gte=updated_at_after))
```

Comment:
interesting that we were able to drop the need for including elements with isnull=False - The reason I initially included them was due to this advisory note in the django docs about how different databases may return different results if you don't filter out null values for that field( https://docs.djangoproject.com/en/1.11/ref/models/querysets/ if you search for "earliest() and latest() may return instances with null dates.")

Reply:
That seems to only be about ordering which is not relevant here.  We are already assuming in the query that the value not existing is causing it to get filtered out in the .after case due to the different types of user profiles so I think things would fail pretty hard in the tests or live if this didn't work and I don't see any evidence of either problem currently.  If we were to switch databases then we might have to address this then, but I would guess not (again since the warning is about ordering not filter/exclude).  Also we specifically don't want to filter out the null values from the .before case.  Finally, the old code wasn't do any of this right anyway. :-)